src/main/java/seedu/pill/util/ItemMap.java

```diff
@@ -0,0 +1,180 @@
+                    + name + ": " + removedItem.getQuantity());
+        } else {
+            LOGGER.warning("Attempt to delete non-existent item: " + name);
+            System.out.println("Item not found: " + name);
+        }
+    }
+
+    /**
+     * Edits an item by its name.
+     *
+     * @param name     The name of the item to be edited.
+     * @param quantity The new quantity for the item.
+     */
+    public void editItem(String name, int quantity) {
+        assert name != null && !name.trim().isEmpty() : "Item name cannot be null or empty";
+        assert quantity > 0 : "Quantity must be positive";
+
+        if (name == null || name.trim().isEmpty() || quantity <= 0) {
+            LOGGER.warning("Attempt to edit item with invalid parameters: name=" + name + ", quantity=" + quantity);
+            System.out.println("Invalid item name or quantity.");
+            return;
+        }
+
+        Item item = items.get(name);
+        if (item != null) {
+            item.setQuantity(quantity);
+            LOGGER.info("Edited item: " + name + ", new quantity: " + quantity);
+            System.out.println("Edited item: " + name + " (" + quantity + " in stock)");
+        } else {
+            LOGGER.warning("Attempt to edit non-existent item: " + name);
+            System.out.println("Item not found: " + name);
+        }
+    }
+
+    /**
+     * Lists all the items in the inventory.
+     */
+    public void listItems() {
+        if (items.isEmpty()) {
+            LOGGER.info("Attempted to list items, but inventory is empty");
+            System.out.println("The inventory is empty.");
+            return;
+        }
+        LOGGER.info("Listing all items in inventory");
+        System.out.println("Listing all items:");
+        int index = 1;
+        for (Map.Entry<String, Item> entry : items.entrySet()) {
+            Item item = entry.getValue();
+            System.out.println(index + ". " + item.getName() + ": " + item.getQuantity() + " in stock");
+            index++;
+        }
+    }
+
+    /**
+     * Finds an item in the list.
+     *
+     * @param itemName The name of the item.
+     */
+    public ItemMap findItem(String itemName) {
+        assert itemName != null : "Item name cannot be null";
+
+        ItemMap foundItems = new ItemMap();
+        if (itemName == null || itemName.trim().isEmpty()) {
+            LOGGER.warning("Attempt to find item with null or empty name");
+            return foundItems;
+        }
+        LOGGER.info("Searching for items containing: " + itemName);
+        for (Map.Entry<String, Item> entry : items.entrySet()) {
+            Item item = entry.getValue();
+            if (item.getName().toLowerCase().contains(itemName.toLowerCase())) {
+                foundItems.addItemSilent(item.getName(), item.getQuantity());
+            }
+        }
+        LOGGER.info("Found " + foundItems.items.size() + " items matching: " + itemName);
+        return foundItems;
```

Comment:
Good job maintaining the usage of ItemList although the implementation was changed